parsec/interfaces/ptg/ptg-compiler/jdf2c.c

```diff
@@ -1579,16 +1588,26 @@ static void jdf_generate_structure(jdf_t *jdf)
         /* If the profile property is ON then enable the profiling array */
         need_profile = profile_enabled(f->properties);
     }
+    /**
+     * not all elements are set to {-1} in this initialization,
+     * but later they will be initialized
+     */
     if( need_profile )
         coutput("#if defined(PARSEC_PROF_TRACE)\n"
                 "#  if defined(PARSEC_PROF_TRACE_PTG_INTERNAL_INIT)\n"
                 "static int %s_profiling_array[4*PARSEC_%s_NB_TASK_CLASSES] = {-1}; /* 2 pairs (begin, end) per task, times two because each task class has an internal_init task */\n"
+                "static int %s_profiling_array_counter_limit = 4*PARSEC_%s_NB_TASK_CLASSES;\n"
                 "#  else /* defined(PARSEC_PROF_TRACE_PTG_INTERNAL_INIT) */\n"
                 "static int %s_profiling_array[2*PARSEC_%s_NB_TASK_CLASSES] = {-1}; /* 2 pairs (begin, end) per task */\n"
+                "static int %s_profiling_array_counter_limit = 2*PARSEC_%s_NB_TASK_CLASSES;\n"
```

Comment:
We shouldn't need `%s_profiling_array_counter_limit` in this structure: AFAIU, it's always `2*nb_task_classes`.

Reply:
or 4 ... depending if we profile the internal startup tasks.